### PR TITLE
SPM improvements: migrating Obj-C to Swift

### PR DESCRIPTION
**Pull request by stury**
Since the previous SPM version didn't really work for Swift apps when consuming the full SDK, I've simplified the Package, and converted all of the code to Swift. 

All Unit Tests run successfully!

There is a breaking change in Swift usage, where the `urlRequest()` methods now return `URLRequest?`, instead of `URLRequest!` as a result. This is technically a breaking change, when being consumed by Swift.  Because of this, we will bump up the major version number.  

Obj-C consumers of this package may have to switch to using the `@import TDOAuth;` notation to import the header, since there is no longer a `TDOAuth.h` file in the repo anymore.  All other aspects of Objective-C consumption are the same as they were.  No changes!

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by master-nevi** [comment]
I'm not a core developer of this library but wouldn’t it make sense to just make it all one `SPM-target` / `pod-subspec` if this change is to be major version bump? `TDOAuth.m/h` seemed to have critical functionality/conveniences that was missing from the `TDOAuth/Swift`. To me that makes it part of the core library. Especially now that it’s all Swift, it doesn’t make too much sense to callout a `TDOAuth/Swift` portion of the library in the podspec or package anymore IMO.

**Comment by stury** [comment]
If that makes sense to other people, I can make that change to merge all of the targets into one.  It makes sense to me. 😄

**Comment by stury** [comment]
As suggested, I reduced the complexity of the `podspec` as well as the SPM package down to one subspec/target respectively.

File: Example/Tests/Compat.m [hunk 1]
```diff
@@ -216,7 +216,8 @@ - (void)testPostUrlParameters
     NSURLRequest *postRequest = [self.class makePostRequestWithDataEncoding:TDOAuthContentTypeUrlEncodedQuery];
 
     NSString *url = [[postRequest URL] absoluteString];
-    XCTAssert([url isEqualToString:@"http://api.example.com/service?foo=bar&baz=bonk"],
+    // In Swift, the order of the key:value pair is undetermined, so either result is correct.
```
Comment (adamkaplan) [inline Example/Tests/Compat.m:219]: Yes, and IIRC the Obj-C version also didn't offer order preservation. It did however offer stability that the tests took advantage of, so this does not indicate changed behavior overall.

**Review by adamkaplan — CHANGES_REQUESTED** [review]
Thanks for fixing this @stury!
Since there is a major bump, please:
1. Remove the new compat code you added (sorry). Don't re-add the obj-c.
2. Note in README obj-c is supported through 1.5.x only.

Contact Yahoo OSWG and ask to be made a core contributor. I can't do that anymore, but you can link to this message.

**Pull request closed without merging**